pkg/api/v1/defaults.go

```diff
@@ -100,9 +100,12 @@ func addDefaultingFuncs() {
 			if obj.RestartPolicy == "" {
 				obj.RestartPolicy = RestartPolicyAlways
 			}
-			if obj.HostNetwork {
+			if obj.HostNetwork || (obj.SecurityContext != nil && obj.SecurityContext.HostNetwork) {
```

Comment:
This is not back-compatible.  If a client only understands the `pod.spec.hostNetwork` flag, this new flag is not going to be handled correctly.

t0: submit a pod with `pod.spec.hostNetwork: true` and `pod.securityContext.hostNetwork: true` - this triggers
t1: user with old client changes `pod.spec.hostNetwork` to `false`
t2: this PR screws up and uses the wrong value.

The old value has to be respected.  Any code that uses the new field ALSO has to use the old field.  Given that, I think this whole PR is pointless, sadly.

I would abandon the idea that we can move fields around - you can;t do it in a compatible way.

File a bug to fix this in v2 and ref the v2 API issue (for tracking) and move on.